metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -79,8 +79,8 @@ class ScalametaLanguageServer(
     compilerConfigPublisher.map(scalac.loadNewCompilerGlobals)
   val scalafixNotifications: Observable[Effects.PublishLinterDiagnostics] =
     metaSemanticdbs.map(scalafix.reportLinterMessages)
-  private var cancelEffects = Option.empty[Cancelable]
-  val effects: Observable[Effects] = Observable.merge(
```

Comment:
cc/ @alexandru can you perhaps explain the difference between `Observable.merge[T](Observable*).subscribe()` and `List[Observable[T])].map(_.subscribe())`? Our tests passed with List but fail with merge, intuitively I would expect them to behave similarly.

Reply:
Don't know what your test is doing, but one thing to be aware of is that `merge` introduces a buffer between the observables pushing the events and the subscriber. And in testing this could mean extra async boundaries, because the consumer is no longer consuming from the same thread as the producer.

Reply:
Interesting, thanks for the explanation. In our tests we use `TestScheduler` and `tickOne` as much as possible https://github.com/scalameta/language-server/blob/617d8644922f379f0d0a874c679aaa04347c92f5/metaserver/src/test/scala/tests/search/SymbolIndexTest.scala#L47  Looking at the logs, it seems the ordering of events is slightly different with/without merge.

Cheers 🍻

Reply:
Ah yes, so ordering can be a problem — `merge` is non-deterministic, creating a race between the participating producers.

Reply:
Our tests don't rely on ordering as far I can tell, but I'll need closer look to verify.

Understanding `merge` better will help me keep an eye out for this in the future. So far Observable[T] has been amazing, this PR was a decent refactoring and all tests passed on first compile 😄  `.merge` causing the tests to fail came as a surprise however, but I wouldn't be surprised if the tests need a refactoring too!